Source/JavaScriptCore/API/tests/FunctionToStringTests.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2023 Colin Vidal <colin@cvidal.org> All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "FunctionToStringTests.h"
+
+#include "InitializeThreading.h"
+#include "JavaScript.h"
+
+int testFunctionToString()
+{
+    const char* inputScript = "var valid = true;"                       "\n"
+        "function foo2   () {}"                                         "\n"
+        "valid &&= foo2.toString() == 'function foo2   () {}'"          "\n"
+        "function       foo3()   {}"                                    "\n"
+        "valid &&= foo3.toString() == 'function       foo3()   {}'"     "\n"
+        "function*  fooGen   (){}"                                      "\n"
+        "valid &&= fooGen.toString() == 'function*  fooGen   (){}'"     "\n"
+        "async function fnasync() {}"                                   "\n"
+        "valid &&= fnasync.toString() == 'async function fnasync() {}'" "\n"
+        "let f1 = async function() {}"                                  "\n"
+        "valid &&= f1.toString() == 'async function() {}'"              "\n"
+        "let f2 = async()=>{}"                                          "\n"
+        "valid &&= f2.toString() == 'async()=>{}'"                      "\n"
+        "let f3 = async  ()    =>     {}"                               "\n"
+        "valid &&= f3.toString() == 'async  ()    =>     {}'"           "\n"
+        "valid";
```

Comment:
I initially wrote this way:

```C++
    const auto inputScript = R"(
        var valid = true;

        function foo2   () {}
        valid &&= foo2.toString() == "function foo2   () {}"

        function       foo3()   {}
        valid &&= foo3.toString() == "function       foo3()   {}"

        function*  fooGen   (){}
        valid &&= fooGen.toString() == "function*  fooGen   (){}"

        async function fnasync() {}
        valid &&= fnasync.toString() == "async function fnasync() {}"

        let f1 = async function() {}
        valid &&= f1.toString() == "async function() {}"

        let f2 = async()=>{}
        valid &&= f2.toString() == "async()=>{}"

        let f3 = async  ()    =>     {}
        valid &&= f3.toString() == "async  ()    =>     {}"

        valid
    )";
```

As it seems clearer to me, but I understand raw string aren't accepted/considered as bad practice in this project?

Reply:
For Raw string, it is fine :)

Reply:
Thank for your comments :-) I wasn't sure how to run those tests, looks like I definitely broke a bunch of legit test cases

[expectations-diff.txt](https://github.com/WebKit/WebKit/files/11669154/expectations-diff.txt)

I'll figure out what's wrong and follow-up. (And also re-add raw string then, plus few extra tests cases and add those in JSTests/stress).

Reply:
Also, probably you need to adjust / modify / verify the test failures in JSTests etc. (as JSC tests are failing).
You can run these tests with `Tools/Scripts/run-javascriptcore-tests` command. https://ews-build.webkit.org/#/builders/12/builds/8723
Maybe, some are real issues. Some are not (need to rebase test expectations).